docs/extend/extend-apiml/api-mediation-routing.md

```diff
@@ -0,0 +1,97 @@
+
+In addition to the basic Zuul routing, the Zowe API Gateway supports versioning in which a major version can be specified. 
+The Gateway routes a request only to an instance that provides the specified major version of the API. 
+The `/api/` prefix is used for REST APIs. The prefix `/ui/` applies to web UIs and the prefix `/ws/` applies to WebSockets.
+
+Additional routing can be implemented using a Zuul pre-filter.
+
+The URL format expected by the API Gateway is:
+
+`https://{gatewayHost}:{port}/api/v{majorVersion}/{serviceId}/{resource}`
+
+**Example:**
+
+The following address shows the original URL of a resource exposed by a service
+```
+http://service:10015/enablerv1sampleapp/api/v1/samples
+```
+The following address shows the API Gateway URL of the resource:
+```
+https://gateway:10010/api/v1/enablerv1sampleapp/samples
+```
+
+The following diagram presents how basic routing works:
+<img src="../../images/api-mediation/Basic-Routing.png" alt="Zowe API Mediation basic routing"/> 
+
+
+### Implementation Details
+
+The service instances provide information about routing to the API Gateway via the Eureka metadata.
+
+**Example:**
+
+    metadata-map:
+        apiml:
+            routes:
+                ui_v1:
+                    gatewayUrl: "ui/v1"
+                    serviceUrl: "/helloworld"
+                api_v1:
+                    gatewayUrl: "api/v1"
+                    serviceUrl: "/helloworld/v1"
+                api_v2:
+                    gatewayUrl: "api/v2"
+                    serviceUrl: "/helloworld/v2"
+
+In this case, the service has a service ID of `helloworldservice` that exposes the following endpoints:
+
+* **UI** - `https://gateway/ui/v1/helloworldservice` routed to `https://hwServiceHost:port/helloworld/`
+* **API major version 1** - `https://gateway/api/v1/helloworldservice` routed to `https://hwServiceHost:port/helloworld/v1`
+* **API major version 2** - `https://gateway/api/v2/helloworldservice` routed to `https://hwServiceHost:port/helloworld/v2`
+
+where:
+
+* The gatewayUrl is matched against the prefix of the URL path used at the Gateway `https://gateway/urlPath` where urlPath is `prefix/serviceId/resourcePath`. 
+* The service ID is used to find the service host and port. 
+* The serviceUrl is used to prefix the resourcePath at the service host.
+
+**Note:** The service ID is not included in the routing metadata, but the service ID is in the basic Eureka metadata. 
+
+## Basic Routing (using only the service ID)
+
+This method of routing is the same as the previous method but does not use the version part of the URL. This is useful for services that handle their versioning themselves with different granularity.
+An example of this is z/OSMF.
+
+**Example:**
+
+z/OSMF URL through the Gateway: `https://gateway:10010/api/zosmf/restjobs/1.0/...`
+
+In this URL example, `zosmf` is the service ID, and `/restjobs/1.0/...` is the rest of the endpoint segment.
+No version is specified in this URL.
```

Comment:
If the 1.0 in this example does not refer to a version number, shouldn't we explain what this does refer to?

Reply:
No it's not related to the API Version, but I agree that it is confusing, and actually this specific z/OSMF endpoint should be /zosmf1/zosmf/restjobs/jobs. It was just an example anyway, but good catch, I can modify it to `https://gateway:10010/api/zosmf/restjobs/jobs/...`